src/lib/mina_base/account_update.ml

```diff
@@ -955,83 +955,52 @@ module Account_precondition = struct
   [%%versioned
   module Stable = struct
     module V1 = struct
-      type t =
-            Mina_wire_types.Mina_base.Account_update.Account_precondition.V1.t =
-        | Full of Zkapp_precondition.Account.Stable.V2.t
-        | Nonce of Account.Nonce.Stable.V1.t
-        | Accept
+      type t = Zkapp_precondition.Account.Stable.V2.t
       [@@deriving sexp, yojson, hash]
 
-      let to_latest = Fn.id
-
-      let to_full = function
-        | Full s ->
-            s
-        | Nonce n ->
-            { Zkapp_precondition.Account.accept with
-              nonce = Check { lower = n; upper = n }
-            }
-        | Accept ->
-            Zkapp_precondition.Account.accept
+      let (_ :
+            ( t
+            , Mina_wire_types.Mina_base.Account_update.Account_precondition.V1.t
+            )
+            Type_equal.t ) =
+        Type_equal.T
 
-      let equal p q = Zkapp_precondition.Account.equal (to_full p) (to_full q)
+      let to_latest = Fn.id
 
-      let compare p q =
-        Zkapp_precondition.Account.compare (to_full p) (to_full q)
+      [%%define_locally Zkapp_precondition.Account.(equal, compare)]
     end
   end]
 
-  [%%define_locally Stable.Latest.(to_full, equal, compare)]
+  [%%define_locally Stable.Latest.(equal, compare)]
 
   let gen : t Quickcheck.Generator.t =
+    (* we used to have 3 constructors, Full, Nonce, and Accept for the type t
+       nowadays, the generator creates these 3 different kinds of values, but all mapped to t
+    *)
+    let open Zkapp_basic in
     Quickcheck.Generator.variant3 Zkapp_precondition.Account.gen
       Account.Nonce.gen Unit.quickcheck_generator
     |> Quickcheck.Generator.map ~f:(function
```

Comment:
Why does this quickcheck generator still need to explicitly generate one of these 3 variants? It seems to make more sense to just quickcheck generate each precondition field separately. The set of generated values will still contain the `nonce` and `accept` cases.

Reply:
It doesn't need to, the goal was to generate the `nonce` and `accept` cases with the same frequency as before. If we generate the components independently, those cases will be much rarer.